src/speculators/utils/loading.py

```diff
@@ -0,0 +1,79 @@
+import json
+from pathlib import Path
+from typing import Any, Optional, Union
+
+import torch
+from huggingface_hub import hf_hub_download
+from loguru import logger
+from safetensors import safe_open
+
+
+def load_model_layers(
+    layer_names: list[str], model_path: str
+) -> Optional[Union[torch.Tensor, dict[str, torch.Tensor]]]:
```

Comment:
Better to make your output types as simple as possible so that users don't need their own if/elif checks to parse the return type. Can you return a single dict that is empty if there aren't any, and a dict with a single key if `len(layer_names)==1`?
```suggestion
) -> dict[str, torch.Tensor]:
```

Reply:
sure!